src/main/java/dk/kb/storage/facade/DsStorageFacade.java

```diff
@@ -396,6 +427,37 @@ private static void loadAndSetChildRelations(DsRecordDto currentRecord, HashSet<
     }
 
     
+    /**
+     * This method will load the local tree around the given record. It will
+     * 1) Load the parent if it exists, and this will be set as parent. Parent will not point down to this child
+     * 2) Load all children and set them as children. The children will not point back to this parent.   
+     * 
+     * @param record The input record with the local tree set     *      
+     */
+    
+    private static void setLocalTreeForRecord(DsRecordDto record) throws SQLException {
+       
+        //Set parent
+        String parentId=record.getParentId();
+        if (parentId != null) {
+            DsRecordDto parent = getRecord(parentId);
+            record.setParent(parent);
+        }
+
+        //Set children     
+         List<String> childrenIds = record.getChildrenIds();
+         List<DsRecordDto> childrenRecords = new ArrayList<DsRecordDto>();
+         record.setChildren(childrenRecords); 
+
+         for (String childrenId : childrenIds) {
+             DsRecordDto child = getRecord(childrenId);
+             record.getChildren().add(child);             
+         }                      
+        //childrenIds.forEach( c -> record.getChildren().add(getRecord(c))); // Just to make Toke happy, but only as a comment instead of the for-loop        
```

Comment:
Halfway there :-)

Frankensteins are ugly. The cleaner solution is to replace the full 7 code-lines with
```Java
        List<DsRecordDto> children = record.getChildrenIds().stream()
                .map(DsStorageFacade::getRecord)
                .collect(Collectors.toList());
        record.setChildren(children);
```
or, since we know that there are no existing child-objects, simply:
```Java
        record.getChildrenIds().stream()
                .map(DsStorageFacade::getRecord)
                .forEach(record::addChildrenItem);
```

Reply:
Fixed. Still need to instantiate empty list. The Dto's are generated with null-values. If the Dto could be manual changed to always have empty list as default instead as null, many checks would not be required in the code.